### PR TITLE
New lace version

### PR DESCRIPTION
**Pull request by lauracabayol**
- Implements the DESI Y1 covariance to weight the emulator
- Adds a new module to estiamte compressed parameters with cosmopower

### CODE REVIEW

File: lace/emulator/nn_emulator.py [hunk 1]
```diff
@@ -108,435 +113,121 @@ def __init__(
...
-                    drop_sim=self.drop_sim,
-                    drop_z=self.drop_z,
-                    z_max=self.z_max,
-                )
+            params = EMULATOR_PARAMS[emulator_label]
+            for key, value in params.items():
+                setattr(self, key, value)
+
+    
+        self.archive, self.training_data = select_training(
```
Comment (jchavesmontero) [inline lace/emulator/nn_emulator.py:146]: Having an archive in each emulator is too much, the memory footprint gets huge when using MPI. Do we really need it? We did not have it in the previous version. Same goes for training_data, do we really need the full training_data when reading an emulator, or it would be enough storing a few things? The main limitation of cup1d right now is the memory footprint

File: lace/emulator/nn_emulator.py [hunk 2]
```diff
@@ -108,435 +113,121 @@ def __init__(
...
-            # make sure that input archive / training data are Nyx sims
-            if self.training_data[0]["sim_label"][:3] != "nyx":
-                raise ValueError(
-                    f"Training data for {emulator_label} are not Nyx sims"
-                )
+        self.kp_Mpc = self.archive.kp_Mpc
 
         _ = self.training_data[0]["k_Mpc"] > 0
         self.kmin_Mpc = np.min(self.training_data[0]["k_Mpc"][_])
-        self._calculate_normalization(archive)
```
Comment (jchavesmontero) [inline lace/emulator/nn_emulator.py:456]: I see that for the normalization you only need the archive, right?

File: lace/emulator/nn_emulator.py [hunk 3]
```diff
@@ -108,435 +113,121 @@ def __init__(
...
+        self._calculate_normalization(self.archive)
 
-        # decide whether to train emulator or read from file
-        if train == False:
-            if self.model_path is None:
-                raise ValueError("If train==False, model path is required.")
+        if not train:
+            self._load_pretrained_model()
+        else:
+            self.train()
```
Comment (jchavesmontero) [inline lace/emulator/nn_emulator.py:170]: train could have as an argument training_data

File: lace/emulator/nn_emulator.py [hunk 4]
```diff
@@ -682,6 +373,36 @@ def _get_training_data_nn(self):
 
         return training_data
 
+    def _get_training_cov(self):
+        if self.emulator_label == "Nyx_alphap_cov":
+            training_cov = []
+            self.Y1_relerr = self._laod_DESIY1_err()
```
Comment (jchavesmontero) [inline lace/emulator/nn_emulator.py:379]: _laod_DESIY1_err?

**Review by jchavesmontero — APPROVED** [review]
